File: core/src/de/project/ice/ecs/components/TextureComponent.java
package de.project.ice.ecs.components;

import de.project.ice.utils.Assets;
import org.jetbrains.annotations.NotNull;

public class TextureComponent implements IceComponent<TextureComponent>
{
    @NotNull
    public Assets.TextureRegionHolder region = new Assets.TextureRegionHolder();

    @Override
    public void reset()
    {
        region = new Assets.TextureRegionHolder();
    }

    @Override
    public void copyTo(TextureComponent copy)
    {
        copy.region = Assets.findRegion(region.name);
    }
}

File: core/src/de/project/ice/ecs/components/TextureComponent.kt
package de.project.ice.ecs.components

import de.project.ice.annotations.Property
import de.project.ice.utils.Assets

class TextureComponent : CopyableIceComponent {
    @Property("The current textureregion of this object")
    var region = Assets.Holder.TextureRegion()

    override fun reset() {
        region = Assets.Holder.TextureRegion()
    }

    override fun copyTo(copy: CopyableIceComponent) {
        if (copy is TextureComponent) {
            copy.region = Assets.findRegion(region.name)
        }
    }
}
